kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoTelemetryListener.java

```diff
@@ -0,0 +1,138 @@
+ */
+public class MongoTelemetryListener implements CommandListener {
+  private static final String MONGODB_METRICS_GROUP = "responsive-mongodb";
+
+  static final MetricName COMMANDS_SUCCEEDED_LATENCY = new MetricName(
+      "commands-succeeded-cumulative-latency",
+      MONGODB_METRICS_GROUP,
+      "cumulative commands succeeded latency",
+      Collections.emptyMap()
+  );
+  private static final MetricName COMMANDS_SUCCEEDED_COUNT = new MetricName(
+      "commands-succeeded-count",
+      MONGODB_METRICS_GROUP,
+      "total commands succeeded",
+      Collections.emptyMap()
+  );
+
+  static final MetricName COMMANDS_FAILED_COUNT = new MetricName(
+        "commands-failed-count",
+        MONGODB_METRICS_GROUP,
+        "total commands failed",
+        Collections.emptyMap()
+    );
+  static final MetricName COMMANDS_FAILED_LATENCY = new MetricName(
+      "commands-failed-cumulative-latency",
+      MONGODB_METRICS_GROUP,
+      "cumulative commands failed latency",
+      Collections.emptyMap()
+  );
+
+  private final ResponsiveMetrics metrics;
+
+  public MongoTelemetryListener(ResponsiveMetrics metrics) {
+    this.metrics = metrics;
+  }
+
+  @Override
+  public void commandSucceeded(final CommandSucceededEvent event) {
+    getOrCreateSensor(event.getCommandName(), true)
+        .record(event.getElapsedTime(TimeUnit.MILLISECONDS));
+  }
+
+  @Override
+  public void commandFailed(final CommandFailedEvent event) {
+    getOrCreateSensor(event.getCommandName(), false)
+        .record(event.getElapsedTime(TimeUnit.MILLISECONDS));
+  }
+
+  private Sensor getOrCreateSensor(
+      String command,
+      boolean isSuccess
+  ) {
+    String sensorName = "mongodb-commands" + "-" + command +
+        (isSuccess? "-succeeded" : "-failed");
+
+    Sensor sensor = metrics.getSensor(sensorName);
+    if (sensor == null) {
+      sensor = metrics.addSensor(sensorName);
+
+      if (isSuccess) {
+        sensor.add(commandsSucceededCount(command), new CumulativeCount());
+        sensor.add(commandsSucceededLatency(command), new CumulativeSum());
+      } else {
+        sensor.add(commandsFailedCount(command), new CumulativeCount());
+        sensor.add(commandsFailedLatency(command), new CumulativeSum());
+      }
+    }
+
+    return sensor;
+  }
+
+  static MetricName commandsSucceededCount(String command) {
+    return metricName(COMMANDS_SUCCEEDED_COUNT, buildTags(command));
+  }
+
+  static MetricName commandsSucceededLatency(String command) {
+    return metricName(COMMANDS_SUCCEEDED_LATENCY, buildTags(command));
+  }
+
+  static MetricName commandsFailedCount(String command) {
+    return metricName(COMMANDS_FAILED_COUNT, buildTags(command));
+  }
+
+  static MetricName commandsFailedLatency(String command) {
+    return metricName(COMMANDS_FAILED_LATENCY, buildTags(command));
+  }
+
+  private static Map<String, String> buildTags(String command) {
+    return Collections.singletonMap("command", command);
```

Comment:
Yeah, I was thinking about that. At first I had a single metric for all commands, but it felt like it would be too noisy to be useful. I imagine our usage of the commands is pretty limited. I was planning on starting up an application and looking at JMX to get a feel for it. I'll do that before we merge here.